src/entities/vision/vision.cpp

```diff
@@ -45,6 +41,9 @@ Vision::~Vision() {
 }
 
 void Vision::initialization() {
+    // Bind and connect socket in network
+    bindAndConnect();
```

Comment:
I moved/changed this function 'cause **every** action with Sockets are sensible in multithreading: now our entities are starting / running in the **World** thread, so the sockets need to be created when the function is being called in the World module, not in the constructor (that is called in **Suassuna**, another thread)
This caused problems while called **delete**, but now is working fine =)